### PR TITLE
Improve query performance with HBase

### LINKED ISSUE #2177: [Bug] Due to a bug in HBase PrefixFilter, the performance of xhop subgraph query has been degraded.
### Bug Type (问题类型)

performance (性能下降)

### Before submit

- [X] 我已经确认现有的 [Issues](https://github.com/apache/hugegraph/issues) 与 [FAQ](https://hugegraph.apache.org/docs/guides/faq/) 中没有相同 / 重复问题 (I have confirmed and searched that there are no similar problems in the historical issue and documents)

### Environment (环境信息)

- Server Version: 1.0.0 (Apache Release Version)
- Backend: HBase
- OS:  CentOS 7.x 
- Data Size:  20000W vertices, 500000W edges 

### Expected & Actual behavior (期望与实际表现)

gremlin:
```
g.V("user_id")
.repeat(out())
.times(2)
.has('USER', 'is_new', eq(1))
.limit(100).dedup().count()
```

API :
```
http://ip:port/gremlin
```

Before Fix: P99 > 100ms  After Fix:P99 < 100ms
HugeSever metrics
image

HBase metrics
image

### PR DESCRIPTION
**Pull request by JackyYangPassion**
fix #2177

### CODE REVIEW

**Review by javeme — DISMISSED** [review]

File: hugegraph-hbase/src/main/java/org/apache/hugegraph/backend/store/hbase/HbaseSessions.java [hunk 1]
```diff
@@ -418,7 +419,11 @@ default R scan(String table, Set<byte[]> prefixes) {
          */
         default R scan(String table, byte[] startRow, boolean inclusiveStart,
                        byte[] prefix) {
+            short value = (short) (((startRow[0] << 8) | (startRow[1] & 0xFF)) +1);
+            byte[] endRow = ByteBuffer.allocate(2).putShort(value).array();
```
Comment (imbajin) [inline hugegraph-hbase/src/main/java/org/apache/hugegraph/backend/store/hbase/HbaseSessions.java:423]: not quiet clear with it，could u explain it more？

and for any of the scan query condition，we always add the end-row？

could u give a case for it（before vs after），and check the ci problems

**Review by imbajin — CHANGES_REQUESTED** [review]

File: hugegraph-hbase/src/main/java/org/apache/hugegraph/backend/store/hbase/HbaseSessions.java (hunk 1 above)
Comment (javeme) [inline hugegraph-hbase/src/main/java/org/apache/hugegraph/backend/store/hbase/HbaseSessions.java:423]: can call `increaseOne()`?

File: hugegraph-hbase/src/main/java/org/apache/hugegraph/backend/store/hbase/HbaseSessions.java [hunk 2]
```diff
@@ -418,8 +419,17 @@ default R scan(String table, Set<byte[]> prefixes) {
...
+            Scan scan = new Scan();
+            if (table.equals("g_oe") || table.equals("g_ie")) {
+                short value = (short) (((startRow[0] << 8) | (startRow[1] & 0xFF)) +1);
+                byte[] endRow = ByteBuffer.allocate(2).putShort(value).array();
+                scan.withStartRow(startRow, inclusiveStart)
+                    .withStopRow(endRow)
+                    .setFilter(new PrefixFilter(prefix));
+            } else {
+                scan.withStartRow(startRow, inclusiveStart)
+                    .setFilter(new PrefixFilter(prefix));
```
Comment (imbajin) [inline hugegraph-hbase/src/main/java/org/apache/hugegraph/backend/store/hbase/HbaseSessions.java:431]: use one line if ≤ 100

File: hugegraph-hbase/src/main/java/org/apache/hugegraph/backend/store/hbase/HbaseSessions.java [hunk 3]
```diff
@@ -418,8 +419,17 @@ default R scan(String table, Set<byte[]> prefixes) {
          */
         default R scan(String table, byte[] startRow, boolean inclusiveStart,
                        byte[] prefix) {
-            Scan scan = new Scan().withStartRow(startRow, inclusiveStart)
-                                  .setFilter(new PrefixFilter(prefix));
+            Scan scan = new Scan();
+            if (table.equals("g_oe") || table.equals("g_ie")) {
```
Comment (javeme) [inline hugegraph-hbase/src/main/java/org/apache/hugegraph/backend/store/hbase/HbaseSessions.java:423]: can we add a method `HbaseTables.Edge.isEdgeTable(table)`

File: hugegraph-hbase/src/main/java/org/apache/hugegraph/backend/store/hbase/HbaseSessions.java [hunk 4]
```diff
@@ -418,8 +419,17 @@ default R scan(String table, Set<byte[]> prefixes) {
...
+                short value = (short) (((startRow[0] << 8) | (startRow[1] & 0xFF)) +1);
```
Comment (javeme) [inline hugegraph-hbase/src/main/java/org/apache/hugegraph/backend/store/hbase/HbaseSessions.java:424]: can we add some comments for why?

**Comment by imbajin** [comment]
@JackyYangPassion address this PR?

**Comment by javeme** [comment]
Is it still in progress?

**Comment by JackyYangPassion** [comment]
Thank you everyone for reviewing and giving this PR #2364 to fix in the future.

**Pull request closed without merging**